src/hotspot/share/c1/c1_LIR.cpp

```diff
@@ -351,7 +351,7 @@ LIR_OpArrayCopy::LIR_OpArrayCopy(LIR_Opr src, LIR_Opr src_pos, LIR_Opr dst, LIR_
   , _expected_type(expected_type)
   , _flags(flags) {
 #if defined(X86) || defined(AARCH64) || defined(S390) || defined(RISCV64) || defined(PPC64)
-  if (expected_type != nullptr && flags == 0) {
+  if (expected_type != nullptr && ((flags & ~LIR_OpArrayCopy::unaligned) == 0)) {
```

Comment:
I was concerned that this is platform-specific, but I checked and all platforms can handle unaligned or overlapping w/o using the stub.  So maybe this should be using LIR_OpArrayCopy::unaligned|LIR_OpArrayCopy::overlapping?

Reply:
Yes, LIR_OpArrayCopy::overlapping was also reset if `OmitChecksFlag` is true. Added  `LIR_OpArrayCopy::overlapping` too.